code/modules/food_and_drinks/restaurant/_venue.dm

```diff
@@ -246,14 +259,14 @@
 
 	to_chat(user, span_notice("You change the portal's linked venue."))
 
-	if(linked_venue && linked_venue.restaurant_portal) //We're already linked, unlink us.
-		if(linked_venue.open)
+	if(linked_venue && (src in linked_venue.restaurant_portals)) //We're already linked, unlink us.
+		linked_venue.restaurant_portals -= src
+		if(linked_venue.open && !length(linked_venue.restaurant_portals))
 			linked_venue.close()
-		linked_venue.restaurant_portal = null
 		linked_venue = null
 
 	linked_venue = chosen_venue
-	linked_venue.restaurant_portal = src
+	linked_venue.restaurant_portals |= src
```

Comment:
why |= if we check contents just above? concerns with existing venue? can we be in more then one? just safety?